modules/seqexec/server/src/main/scala/seqexec/server/gmos/GmosKeywordReader.scala

```diff
@@ -26,18 +32,66 @@ final case class GmosObsKeywordsReader[F[_]: Sync](config: Config) {
           .getOrElse(YesNoType.NO))
       .map(_.toBoolean)
       .safeValOrDefault
+
+  def nodMode: F[String] = "STANDARD".pure[F]
+
+  def nodPix: F[Int] = Gmos.nodAndShuffle(config).map(_.rows).explainExtractError
+
+  def nodCount: F[Int] = Gmos.nodAndShuffle(config).map(_.cycles).explainExtractError
+
+  def nodAxOff: F[Double] = Gmos.nodAndShuffle(config).explainExtractError
+    .flatMap(
+      _.positions.find(_.stage === StageA)
+        .map(x => (x.offset.p.toAngle.toMicroarcseconds.toDouble/1e6).pure[F])
+        .getOrElse(SeqexecFailure.Unexpected("Cannot find stage A parameters in step configuration.")
+          .raiseError[F, Double]
+        )
+    )
+
+  def nodAyOff: F[Double] = Gmos.nodAndShuffle(config).explainExtractError
+    .flatMap(
+      _.positions.find(_.stage === StageA)
+        .map(x => (x.offset.q.toAngle.toMicroarcseconds.toDouble/1e6).pure[F])
+        .getOrElse(SeqexecFailure.Unexpected("Cannot find stage A parameters in step configuration.")
+          .raiseError[F, Double]
+        )
+    )
+
+  def nodBxOff: F[Double] = Gmos.nodAndShuffle(config).explainExtractError
+    .flatMap(
+      _.positions.find(_.stage === StageB)
+        .map(x => (x.offset.p.toAngle.toMicroarcseconds.toDouble/1e6).pure[F])
+        .getOrElse(SeqexecFailure.Unexpected("Cannot find stage A parameters in step configuration.")
+          .raiseError[F, Double]
+        )
+    )
+
+  def nodByOff: F[Double] = Gmos.nodAndShuffle(config).explainExtractError
+    .flatMap(
+      _.positions.find(_.stage === StageB)
+        .map(x => (x.offset.q.toAngle.toMicroarcseconds.toDouble/1e6).pure[F])
+        .getOrElse(SeqexecFailure.Unexpected("Cannot find stage A parameters in step configuration.")
+          .raiseError[F, Double]
+        )
+    )
+
+  def isNS: F[Boolean] = config.extractInstAs[java.lang.Boolean](USE_NS_PROP).map(_.booleanValue).explainExtractError
+
+}
+
+object GmosObsKeywordsReader {
+
+  private implicit class ExplainExtractError[F[_]: Sync, A](v: Either[ExtractFailure, A]) {
```

Comment:
Do you need `Sync` here? I think `MonadError` is enough

Reply:
Y tried with `MonadError` first, but then I got compilation errors about ambiguous implicits.

Reply:
That may mean you are missing to explicity set the `F`. e.g. `widenRethrowT[F]`?

Reply:
I rearranged the type parameters and put some explicitly, and now it compiles with `MonadError`